### ISSUE TITLE
Breeze throws exception when querying entities with array of complexTypes (isScalar: false)

### ISSUE THREAD

**Issue by swiety85** [Breeze/breeze.js#183]
After upgrade Breeze version to 1.5.4 I've got an error while I'm loading entities that has list of nested complexType objects:
`TypeError: Cannot read property 'parentProperty' of undefined at coEquals (/Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:8097:38) at /Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:8103:16 at Array.every (native) at coEquals (/Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:8098:30) at __arrayEquals (/Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:382:14) at /Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:8045:18 at Array.forEach (native) at EntityType.proto._updateTargetFromRaw (/Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:8026:25) at updateEntity (/Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:15450:16) at mergeEntity (/Volumes/Secured/www/breezePlayground/node_modules/breeze-client/build/breeze.debug.js:15397:11)`

As proof of this error you can try to run test from task: Breeze issue with complexType change tracking

I've figured out that "coEquals" recursive method tries to be used on complexArrays. The problem is that it works only with complexTypes. I wrote a quick fix in Breeze code for this function and it seam to work as expected now (but it is only my temporary solution). From that reason I would like to ask for help and solve this issue natively (in Breeze) or provide some workaround solution.

My fix:
`
function coEquals(co1, co2) {
  var dataProps = co1.complexAspect.parentProperty.dataType.dataProperties;
  var areEqual = dataProps.every(function (dp) {
    if (!dp.isSettable) return true;
    var v1 = co1.getProperty(dp.name);
    var v2 = co2.getProperty(dp.name);
    if (dp.isComplexProperty && dp.isScalar) {
      return coEquals(v1, v2);
    }
    else if(dp.isComplexProperty && !dp.isScalar) {
      return __arrayEquals(v1, v2, coEquals)
    } else {
      var dataType = dp.dataType; // this will be a complexType when dp is a complexProperty
      return (v1 === v2 || (dataType && dataType.isDate && v1 && v2 && v1.valueOf() === v2.valueOf()));
    }
  });
  return areEqual;
}
`

**Comment by AdamOke** [comment]
We just upgraded to angular 1.6, which required an update to breeze 1.6.1 (we moved to the newest available as of this writing which is breeze 1.6.2) and now we are hitting this same problem. We use arrays of complex types pretty extensively - applying your recommended change to the library seemed to resolve the problem without any negative side effects. Thanks for the post - this should work as a short-term solution until an updated library is available.
